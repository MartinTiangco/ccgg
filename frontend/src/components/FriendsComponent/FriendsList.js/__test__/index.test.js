import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import React from "react";
import FriendsList from "../index";

const dummyFriends = [
  { username: "Username1", summonerName: "bob1111112222" },
  { username: "Username2", summonerName: "coolname" },
  { username: "Username3", summonerName: "lucas" },
  { username: "Username4", summonerName: "robot" },
  { username: "Username5", summonerName: "suffering" },
  { username: "Username6", summonerName: "awaits" },
];

const emptyFriends = [];

it("snapshot with dummy friend data", () => {
  const tree = renderer.create(<FriendsList friends={dummyFriends} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("snapshot with empty friend data", () => {
  const tree = renderer.create(<FriendsList friends={emptyFriends} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Check multiple friends rendered correctly", () => {
  render(<FriendsList friends={dummyFriends} />);

  // Check all 6 different friends are rendered
  expect(screen.getByText("Username1")).toBeInTheDocument();
  expect(screen.getByText("Username2")).toBeInTheDocument();
  expect(screen.getByText("Username3")).toBeInTheDocument();
  expect(screen.getByText("Username4")).toBeInTheDocument();
  expect(screen.getByText("Username5")).toBeInTheDocument();
  expect(screen.getByText("Username6")).toBeInTheDocument();

  // Check all 6 ping buttons/images are rendered
  expect(screen.getAllByAltText("Ping Icon")).toHaveLength(6);

  // Check that only 6 rows are generated
  expect(screen.getAllByText(/Summoner:/)).toHaveLength(6);

  // Check search bar is rendered correctly
  expect(screen.getAllByLabelText("Enter username...")).toHaveLength(1);
  expect(screen.getByLabelText("Enter username...")).toBeInTheDocument();

  // Check that add friend button + tool tip is rendered
  expect(screen.getByTitle("Add Friend")).toBeInTheDocument();
});

it("Check no friends rendered correctly", () => {
  render(<FriendsList friends={emptyFriends} />);

  // Check no friends component is rendered correctly
  expect(screen.getByText("No friends found!")).toBeInTheDocument();

  // Check search bar is rendered correctly
  expect(screen.getAllByLabelText("Enter username...")).toHaveLength(1);
  expect(screen.getByLabelText("Enter username...")).toBeInTheDocument();

  // Check that add friend button + tool tip is rendered
  expect(screen.getByTitle("Add Friend")).toBeInTheDocument();
});
