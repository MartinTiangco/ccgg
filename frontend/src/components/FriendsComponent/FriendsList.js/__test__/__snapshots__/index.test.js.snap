// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot with dummy friend data 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-add-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-11"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          onInput={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root makeStyles-textFieldColor-5 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="outlined-basic"
            id="outlined-basic-label"
          >
            Enter username...
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root makeStyles-textFieldColor-5 MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="outlined-basic"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline makeStyles-noBorder-4"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Enter username...
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-1"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Add Friend"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-scroll-8 MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-12"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-friendRow-6 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              Username1
            </h5>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-7"
          >
            <p
              className="MuiTypography-root makeStyles-summonerName-1 MuiTypography-body2"
            >
              Summoner: 
              bob1111112222
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-describedby={null}
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Ping Friend"
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="Ping Icon"
                className="makeStyles-iconImg-2"
                src="missing.png"
              />
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-friendRow-6 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              Username2
            </h5>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-7"
          >
            <p
              className="MuiTypography-root makeStyles-summonerName-1 MuiTypography-body2"
            >
              Summoner: 
              coolname
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-describedby={null}
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Ping Friend"
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="Ping Icon"
                className="makeStyles-iconImg-2"
                src="missing.png"
              />
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-friendRow-6 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              Username3
            </h5>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-7"
          >
            <p
              className="MuiTypography-root makeStyles-summonerName-1 MuiTypography-body2"
            >
              Summoner: 
              lucas
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-describedby={null}
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Ping Friend"
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="Ping Icon"
                className="makeStyles-iconImg-2"
                src="missing.png"
              />
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-friendRow-6 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              Username4
            </h5>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-7"
          >
            <p
              className="MuiTypography-root makeStyles-summonerName-1 MuiTypography-body2"
            >
              Summoner: 
              robot
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-describedby={null}
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Ping Friend"
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="Ping Icon"
                className="makeStyles-iconImg-2"
                src="missing.png"
              />
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-friendRow-6 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              Username5
            </h5>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-7"
          >
            <p
              className="MuiTypography-root makeStyles-summonerName-1 MuiTypography-body2"
            >
              Summoner: 
              suffering
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-describedby={null}
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Ping Friend"
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="Ping Icon"
                className="makeStyles-iconImg-2"
                src="missing.png"
              />
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-friendRow-6 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-11"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-4"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5"
            >
              Username6
            </h5>
          </div>
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-7"
          >
            <p
              className="MuiTypography-root makeStyles-summonerName-1 MuiTypography-body2"
            >
              Summoner: 
              awaits
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-1"
        >
          <button
            aria-describedby={null}
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            title="Ping Friend"
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="Ping Icon"
                className="makeStyles-iconImg-2"
                src="missing.png"
              />
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot with empty friend data 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-add-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-11"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          onInput={[Function]}
          onKeyPress={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root makeStyles-textFieldColor-5 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="outlined-basic"
            id="outlined-basic-label"
          >
            Enter username...
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root makeStyles-textFieldColor-5 MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="outlined-basic"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline makeStyles-noBorder-4"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Enter username...
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-1"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Add Friend"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-scroll-8 MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      No friends found!
    </div>
  </div>
</div>
`;
